content/posts/2023-05-15-mac-learning.md

```diff
@@ -0,0 +1,303 @@
+02:11:11:11:00:00 dev eth0 master br0 
+02:22:22:22:00:00 dev eth1 master br0 
+02:33:33:33:00:00 dev eth2 master br0    <=== NEW (H3)
+02:44:44:44:00:00 dev eth2 master br0    <=== NEW (H4)
+02:bb:bb:bb:00:00 dev eth2 master br0
+root@bridge-a:~# 
+
+Bridge-B:
+root@bridge-b:~# bridge fdb show dynamic
+02:22:22:22:00:00 dev eth0 master br0    <=== NEW (H2)
+02:11:11:11:00:00 dev eth0 master br0 
+02:aa:aa:aa:00:02 dev eth0 master br0 
+02:33:33:33:00:00 dev eth1 master br0 
+02:44:44:44:00:00 dev eth2 master br0 
+root@bridge-b:~# 
+```
+
+The figure below illustrates the MAC databases (FDBs) by using
+symbolic names for the MAC addresses.
+```
+   Bridge-A BaseMAC               Bridge-B BaseMAC
+   02:aa:aa:aa:00:00              02:bb:bb:bb:00:00
+   .----------------------.       .-----------------------.
+   |        Bridge-A      |       |       Bridge-B        |
+   | eth0    eth1    eth2 |       | eth0    eth2     eth2 |
+   '--+-------+-------+---'       '--+-------+--------+---'
+    h1|     h2|   br-b|              |br-a   |h3      |h4    <== FDB
+      |       |     h3'--------------'h1     |        | 
+	  |     .-+--.	h4                h2  .--+-.      |
+	  |     | H2 |	                      | H3 |      |
+	  |     '----'	                      '----'      |
+   .--+-.  IP: 10.0.1.2/24          IP: 10.0.1.3/24 .-+--.
+   | H1 | MAC: 02:22:22:00:00  MAC: 02:33:33::00:00 | H4 |
+   '----'                                           '----'
+ IP: 10.0.1.1/24                            IP: 10.0.1.1/24
+MAC: 02:11:11:11:00:00               MAC: 02:44:44:44::00:00
+```
+
+## Exploring MAC learning with Infix
+
+The examples above were done using a virtual setup with [Infix][Infix] on
+[GNS3][GNS3]. Below are some hints if you like to explore this
+yourself.
+
+To install Infix and GNS3, look at the guide in the  [Infix][Infix]
+README. It is also recommended to look at the [Basic Networking
+Training Post][Training-Basic-Net-Post].
+
+In the setup used here, all units (Bridge-A, Bridge-B, H1, H2, H3 and
+H4) were using _Infix-Classic_ as GNS3 appliance.
+
+### Hints on configuring the Bridges
+
+Bridges-A and Bridges-B were configured with multiple ports (at
+least 3). This is done in the _GNS3 Configure_ window for the units.
+
+To make Bridge-A (and Bridge-B) act like MAC bridges, the following
+commands where issued via the console.
+
+```
+root@bridge-a:~# ip link add br0 type bridge
+root@bridge-a:~# ip link set eth0 master br0
+root@bridge-a:~# ip link set eth1 master br0
+root@bridge-a:~# ip link set eth2 master br0
+root@bridge-a:~# ip link set br0 up
+```
+
+To make Bridge-A get hostname (and prompt) 'bridge-a', you can edit
+'/etc/hostname'. To avoid that Bridge-A and Bridge-B sends a lot of
+DHCP messages, you can comment out the associated line in
+'/etc/network/interfaces.d/ for eth0, eth1, etc.
+
+```
+root@bridge-a:~# cat /etc/network/interfaces.d/eth0 
+auto eth0
+#iface eth0 inet dhcp
+    pre-up ip link set eth0 group iface
```

Comment:
Should recommend commenting-out all three lines.

Reply:
If I remove the, I will have to do "ip link set eth0 up".
Do you think it is ok if I keep that (while removing the other two)?

Reply:
Aha, use inet manual then.

Reply:
Was a bit tired when I read it, the line `pre-up ip link set eth0 group iface` should probably be there, then it gets listed by "show iface".  But you can fix that later, as we discussed AFGH :-)